src/components/selectors/MimicSelector.tsx

```diff
@@ -50,13 +50,19 @@ const MimicSelector: React.FC<MimicSelectorProps> = ({ MimicData, mimicNumber, o
     return value != null && value.length > 0;
   };
 
+  React.useEffect(() => {
+    let numEmptyField = Object.keys(MimicData).length - 1; // minus gameId in MimicData
+    for (const [key, value] of Object.entries(MimicData)) {
+      if (key === 'gameId') continue;
+      if (isFieldValid(value as string)) numEmptyField -= 1;
+    }
+    if (numEmptyField < Object.keys(MimicData).length - 1) setIsError(true);
+  }, []);
+
```

Comment:
Let's try to avoid missing dependencies in UseEffect hooks.

Also, I believe this effect is not useful anymore. Now that the error is not blocking, we can rename the `setIsError` variable to `showError`. The goal would be to show the potential errors only if the user navigates back to this step (or mimic here). Ex:
1. User go to mimic/1. `showError = false`.
2. User navigate to mimic/2. `showError = false`. (for mimic 2)
3. User go back to mimic/1. `showError = true` -> we show were could be the errors.


**Suggestion:**
```tsx
const [showError, setShowError] = React.useState<boolean>(false);

[...]

React.useEffect(() => {
    // if user navigated to the next step already, let's show the errors if there are any.
    if (window.sessionStorage.getItem(`mimic-step-${mimicNumber}-next`) === 'true') {
        setShowError(true);
        window.sessionStorage.setItem(`mimic-step-${mimicNumber}-next`, 'false');
    }
}, []);

[...]

const nextPage = () => {
    save().catch(console.error);
    // save in local storage that the user is going to the next step.
    window.sessionStorage.setItem(`mimic-step-${mimicNumber}-next`, 'true');
    onNext?.();
};

[...]

<TextField
    {...}
    error={showError && !isFieldValid()}
    helperText={showError && !isFieldValid() ? 'Ce champ est obligatoire' : ''}
/>
```

Reply:
Thanks for your review. I have a comment about the useEffect because we should try avoid missing dependencies but in the example here we do not have any. Vscode complained by suggesting to add a dependency on mimicnumber which I did. 
Is this correct here ? Because I thought without dependencies the useEffect will render only once.